js/coinflex.js

```diff
@@ -0,0 +1,1920 @@
+        //          "price": "4.000",
+        //          "quantity": "3.0",
+        //          "side": "BUY",
+        //          "marketCode": "BAND-USD",
+        //          "timeInForce": "GTC",
+        //          "orderType": "LIMIT"
+        //     }
+        //
+        const data = this.safeValue (response, 'data', []);
+        const firstResult = this.safeValue (data, 0, {});
+        const success = this.safeValue (firstResult, 'success');
+        if (success === 'false') {
+            const message = this.safeString (firstResult, 'message');
+            const code = this.safeString (firstResult, 'code');
+            const body = this.id + ' ' + this.json (response);
+            this.throwExactlyMatchedException (this.exceptions['exact'], code, body);
+            this.throwBroadlyMatchedException (this.exceptions['broad'], message, body);
+            throw new ExchangeError (body);
+        }
+        return this.parseOrder (firstResult, market);
+    }
+
+    async cancelAllOrders (symbol = undefined, params = {}) {
+        await this.loadMarkets ();
+        let market = undefined;
+        const request = {};
+        let method = 'privateDeleteV2CancelOrders';
+        if (symbol !== undefined) {
+            market = this.market (symbol);
+            request['marketCode'] = market['id'];
+            method = 'privateDeleteV2CancelOrdersMarketCode';
+        }
+        const response = await this[method] (this.extend (request, params));
+        //
+        //     {
+        //         "event": "orders",
+        //         "timestamp": "1651651758625",
+        //         "accountId": "38420",
+        //         "data": {
+        //             "marketCode": "XRP-USD",
+        //             "msg": "All open orders for the specified market have been queued for cancellation"
+        //         }
+        //     }
+        //
+        // Note: when fired without symbol, the 'data' object only contains a msg property with '{"msg":"All open orders for the account have been queued for cancellation"}'
+        // if there has been no orders pending, then data property will be null.
+        //
+        return response;
+    }
+
+    sign (path, api = 'public', method = 'GET', params = {}, headers = undefined, body = undefined) {
+        const [ finalPath, query ] = this.resolvePath (path, params);
+        let url = this.urls['api'][api] + '/' + finalPath;
+        let encodedParams = '';
+        if (Object.keys (query).length) {
+            encodedParams = this.urlencode (query);
+            if (method === 'GET') {
+                url += '?' + encodedParams;
+            }
+        }
+        if (api === 'private') {
+            this.checkRequiredCredentials ();
+            const nonce = this.nonce ();
+            const datetime = this.ymdhms (this.milliseconds (), 'T');
+            let auth = datetime + '\n' + nonce + '\n' + method + '\n' + this.options['baseApiDomain'] + '\n' + '/' + finalPath + '\n';
```

Comment:
Currently signing is broken on the PHP side. After some digging I noticed that these single-quoted line breaks are the culprit, PHP does not handle them properly. To fix it you should replace this line by:

```Javascript
let auth = datetime + "\n" + nonce + "\n" + method + "\n" + this.options['baseApiDomain'] + "\n" + '/' + finalPath + "\n" + encoded; // eslint-disable-line quotes
```

![image](https://user-images.githubusercontent.com/43336371/166698928-6b832abb-8154-46b9-b3eb-71a5f451dd62.png)

Reply:
right. However, as a second thought, this is not first case of this issue - what do you think, maybe for a long-run and conveniency, we might transpile `'n'` into php with `"\n"`, or maybe we can defined `this.newLine` in base to handle the new line character across different langs and avoid using `\n`s at all. just thinking for a long-run usability.